core/app/assets/stylesheets/refinery/sections/_menu_item_icons.scss

```diff
@@ -0,0 +1,9 @@
+.site-name {@include icon('bars', $plugin_icon_colour, $margin: 1.2em)}
+
+.refinery_images_icon {@include icon('photo', $plugin_icon_colour, $margin: 1.2em)}
```

Comment:
First: although I'll argue against your points @bricesanchez, I'll actually do what you've suggested.

I think that adding a s/css file and modifying a generator is not a big problem either initially or in maintenance.  

Still trying to work out how to do it your way (within the plugin definition) and my way (without adding yet another set of classes.)